phpBB/phpbb/members/controller/online.php

```diff
@@ -0,0 +1,618 @@
+	}
+
+	protected function get_location($session_page, $forum_id) // $forum_id can be removed in the future i think https://tracker.phpbb.com/browse/PHPBB3-15434
+	{
+		global $phpbb_container, $phpbb_root_path, $phpEx, $phpbb_adm_relative_path;
+
+		try
+		{
+			$session_page = parse_url($session_page,  PHP_URL_PATH); // Remove query string
+			$session_page = '/' . ((substr($session_page, 0, 8) == 'app.php/') ? substr($session_page, 8) : $session_page); // Remove app.php/
+
+			$match = $this->router->match($session_page);
+
+			switch ($match['_route'])
+			{
+				case 'phpbb_help_bbcode_controller':
+				case 'phpbb_help_bbcode_controller':
+					$location = $this->language->lang('VIEWING_FAQ');
+					$location_url = $this->helper->route('phpbb_help_faq_controller');
+				break;
+
+				case 'phpbb_members_online':
+				case 'phpbb_members_online_whois':
+					$location = $this->language->lang('VIEWING_ONLINE');
+					$location_url = $this->helper->route('phpbb_members_online');
+				break;
+
+				case 'phpbb_report_pm_controller':
+				case 'phpbb_report_post_controller':
+					$location = $this->language->lang('REPORTING_POST');
+					$location_url = append_sid("{$phpbb_root_path}index.$phpEx");
+				break;
+
+				case 'phpbb_message_admin':
+				//case 'phpbb_message_topic':
+				//case 'phpbb_message_user':
+					$location = $this->language->lang('VIEWING_CONTACT_ADMIN');
+					$location_url = append_sid("{$phpbb_root_path}memberlist.$phpEx", 'mode=contactadmin');
+				break;
+
+				case 'phpbb_members_profile':
+					$location_url = append_sid("{$phpbb_root_path}memberlist.$phpEx");
+					$location = $this->language->lang('VIEWING_MEMBER_PROFILE');
+				break;
+
+				case 'phpbb_members_team':
+					$location_url = append_sid("{$phpbb_root_path}memberlist.$phpEx");
+					$location = $this->language->lang('VIEWING_MEMBERS');
+				break;
+
+				default:
+					// Is a route, but not in the switch
+					$location = $this->language->lang('INDEX');
+					$location_url = append_sid("{$phpbb_root_path}index.$phpEx");
+			}
+		}
+		catch (\RuntimeException $e) // Urls without route
+		{
+			/* @var $viewonline_helper \phpbb\viewonline_helper */
+			$viewonline_helper = $phpbb_container->get('viewonline_helper');
+			$on_page = $viewonline_helper->get_user_page($session_page);
+
+			switch ($on_page[1])
+			{
+				case 'index':
+					$location = $this->language->lang('INDEX');
+					$location_url = append_sid("{$phpbb_root_path}index.$phpEx");
+				break;
+
+				case $phpbb_adm_relative_path . 'index':
+					$location = $this->language->lang('ACP');
+					$location_url = append_sid("{$phpbb_root_path}index.$phpEx");
+				break;
+
+				case 'posting':
+				case 'viewforum':
+				case 'viewtopic':
+
+					if ($forum_id && $this->auth->acl_get('f_list', $forum_id))
+					{
+						$location = '';
+						$location_url = append_sid("{$phpbb_root_path}viewforum.$phpEx", 'f=' . $forum_id);
+
+						if ($forum_data[$forum_id]['forum_type'] == FORUM_LINK)
```

Comment:
I think you need to move the assignment of `$forum_data` as it's now undefined here.

Reply:
Yes